### PR TITLE
Xplat support for Xamarin Test Cloud (and other Xamarin improvements)

### PR DESCRIPTION
**Pull request by DavidStaheli**

### CODE REVIEW

File: Tasks/XamariniOS/xamariniOS.js [hunk 1]
```diff
@@ -30,21 +30,40 @@ if (!fs.existsSync(mdtoolPath)) {
...
-mdtoolRunner.exec()
-.then(function (code) {
-    // Executed successfully
-    tl.exit(code);
-})
-.fail(function (err) {
+// Find location of nuget
+var nugetPath = tl.which('nuget');
+if (!nugetPath) {
+    tl.error('nuget was not found in the path.');
```
Comment (jsauvexamarin) [inline Tasks/XamariniOS/xamariniOS.js]: Executing `which nuget` on my build agent reveals the path `/usr/local/bin/nuget`. Why would I be getting the error message `nuget was not found in the path`? Is something other than the standard nuget path being passed?

Comment (ericsciple) [inline Tasks/XamariniOS/xamariniOS.js]: Are you running your agent interactively, or as a service?

Comment (jsauvexamarin) [inline Tasks/XamariniOS/xamariniOS.js]: Re-tried with agent version vsts-agent-osx.10.11-x64-2.101.0.tar.gz. Same result: `nuget was not found in the path.`

Comment (bryanmacfarlane) [inline Tasks/XamariniOS/xamariniOS.js]: Is nuget in your path from your terminal?
Are you running interactively or a service?  (Eric's question)
If a service, have you tried .Path and restart? https://github.com/Microsoft/vsts-agent/blob/master/docs/start/nixsvc.md

Comment (jsauvexamarin) [inline Tasks/XamariniOS/xamariniOS.js]: @ericsciple: as a service.

Comment (bryanmacfarlane) [inline Tasks/XamariniOS/xamariniOS.js]: OK.  Look at the .Path and restart link I referred to above.  Also do the comparison.

Finally, this code isn't the best place to track down your issue.  If my link and suggestion above doesn't fix, log an issue

Comment (jsauvexamarin) [inline Tasks/XamariniOS/xamariniOS.js]: Cool, will do. I'll check it out, and report back with an issue. Thx!

**Pull request merged**